### PR TITLE
Add top and right theme settings to axis.ticks and axis.line

### LINKED ISSUE #1899: Need theme setting for right/top ticks
So you can do (e.g.)

```
ggplot(mpg, aes(displ, hwy)) +
  geom_point() +
  geom_hline(yintercept = 235 / l_breaks, colour = "pink") +
  scale_y_continuous(
    "mpg",
    sec.axis = sec_axis(~ 235 / ., name = "l / 100 km", breaks = l_breaks)
  ) +
  theme(
    axis.text.y.right = element_text(colour = "pink"), 
    axis.ticks.y.right = element_line(colour = "pink")
  )
```

### PR DESCRIPTION
**Pull request by thomasp85**
Fixes #1899

### CODE REVIEW

File: R/guides-axis.r [hunk 1]
```diff
@@ -47,24 +47,24 @@ guide_axis <- function(at, labels, position = "right", theme) {
     left =  element_render(theme, label_render, labels, y = label_y, expand_x = TRUE))
 
   line <- switch(position,
-    top =    element_render(theme, "axis.line.x", c(0, 1), c(0, 0), id.lengths = 2),
+    top =    element_render(theme, "axis.line.x.top", c(0, 1), c(0, 0), id.lengths = 2),
```
Comment (hadley) [inline R/guides-axis.r]: I do wonder if we should add `axis.line.x.bottom` for symmetry

Comment (thomasp85) [inline R/guides-axis.r]: yeah, I thought about that too as I revised it now... My sentiment is that the top and right are special cases that deserves a qualifier, while it is expected that `axis.*.x` defaults to bottom etc.

Comment (hadley) [inline R/guides-axis.r]: It makes it inconsistent though with `axis`, `axis.x` etc

Comment (thomasp85) [inline R/guides-axis.r]: sure, And I'm happy to oblige :-)

Comment (thomasp85) [inline R/guides-axis.r]: Is that your wish?

Comment (hadley) [inline R/guides-axis.r]: Yes please!

File: tests/figs/deps.txt [hunk 2]
```diff
@@ -1,3 +1,3 @@
-vdiffr: 0.1.1.9000
+vdiffr: 0.1.1
```
Comment (karawoo) [inline tests/figs/deps.txt]: Is there a particular reason to go back to 0.1.1?

Comment (thomasp85) [inline tests/figs/deps.txt]: No idea how this has snug in...

**Comment by hadley** [comment]
@karawoo could you please merge this by hand? (making sure to eliminate the spurious vdiffr change)

**Pull request merged**